src/ray/raylet/raylet.cc

```diff
@@ -69,6 +69,21 @@ Raylet::Raylet(boost::asio::io_service &main_service, const std::string &socket_
   self_node_info_.set_object_manager_port(object_manager_.GetServerPort());
   self_node_info_.set_node_manager_port(node_manager_.GetServerPort());
   self_node_info_.set_node_manager_hostname(boost::asio::ip::host_name());
+
+  if (RayConfig::instance().gcs_service_enabled()) {
+    RAY_LOG(INFO) << "Gcs service enable.";
+    gcs_client_ = std::make_shared<gcs::ServiceBasedGcsClient>(gcs_client->GetOptions());
+
+    thread_io_service_.reset(new std::thread([this] {
+      std::unique_ptr<boost::asio::io_service::work> work(
+          new boost::asio::io_service::work(io_service_));
+      io_service_.run();
+    }));
+
+    RAY_CHECK_OK(gcs_client_->Connect(io_service_));
+  } else {
+    gcs_client_ = gcs_client;
+  }
```

Comment:
Consolidate this with the gcs client initialization code in `main.cc`.

Reply:
fixed